### PR TITLE
Add goals to peer draining

### PR DESCRIPTION
**Pull request by jcorbin**
Easy to accomodate now that we have the explicit `PeerDrain` object.  As
mentioned in early review for peer drain, we always wanted to have the ability
to say "and then ...".

This adds the two "and then ..."s that are used in Hyperbahn now/soon.

r @raynos @rf

### CODE REVIEW

File: drain.js [hunk 1]
```diff
@@ -142,6 +172,62 @@ function stop() {
...
+
+    var info = self.peer.extendLogInfo(self.extendLogInfo({
+        error: err
+    }));
+
+    if (err.type === 'tchannel.drain.peer.timed-out') {
+        self.peer.logger.warn(
+            'drain timed out, force closing connections',
+            info);
+    } else {
```
Comment (Raynos) [inline drain.js]: `else if err` ?

File: drain.js [hunk 2]
```diff
@@ -142,6 +172,62 @@ function stop() {
...
+
+    var info = self.peer.extendLogInfo(self.extendLogInfo({
+        error: err
+    }));
+
+    if (err.type === 'tchannel.drain.peer.timed-out') {
+        self.peer.logger.warn(
+            'drain timed out, force closing peer',
+            info);
+    } else {
```
Comment (Raynos) [inline drain.js]: `else if err` ?

**Comment by Raynos** [comment]
lgtm.

**Comment by jcorbin** [comment]
Tested by `npm link` locally in https://github.com/uber/hyperbahn/pull/130

**Pull request merged**